### DISCUSSION TITLE
What encryption is used for Android client ID ?

### DISCUSSION THREAD

**Discussion by SoiGitRepo** [rustdesk/rustdesk#4706] [General]
I deployed my own server, and I want to forcefully change RustDesk client ID on android client, just found RustDesk.toml file, then I saw that ID was stored as encrypted string, can i just replace that encrypted string to change the ID? and i wonder what encryption method is used.

id in toml file looks like this
`enc_id = '00hU/ukyJnxSHGoqIPGdxjAW9vU4kmML10p9w='`

**Comment by SergeyMy** [comment]
1. client ID is encrypted based on uuid (unique user id). 
//on server(view database)
`sqlite3 -column db_v2.sqlite3 "select id,created_at,user,note,info,status,uuid,hex(guid) from peer;"`
3. it is enough to build your server from the source code, and by changing one line - allow changing the ID :)

**Reply by quyvsquy** [reply]
> it is enough to build your server from the source code, and by changing one line - allow changing the ID :)
how to do that with the selfhost server? please help me

**Reply by SergeyMy** [reply]
`rustdesk-server/src/rendezvous_server.rs`
LINE 550

options are described in
`rustdesk-server/libs/hbb_common/protos/rendezvous.proto`
LINE 63

PS
but everything you do you do at your own peril and risk.
what you don't do, you don't do.
